Tools/clinic/libclinic/dsl_parser.py

```diff
@@ -1102,6 +1102,9 @@ def bad_node(self, node: ast.AST) -> None:
                     fail("A 'defining_class' parameter cannot have a default value.")
                 if self.group:
                     fail("A 'defining_class' parameter cannot be in an optional group.")
+                assert isinstance(self.block.signatures[-1], Function)
+                if self.block.signatures[-1].cls is None:
```

Comment:
Can't you just check `self.function.cls`?

Reply:
Applied. There was no need to check the mypy.  Am I missing something else?

Reply:
Yes, mypy should be fine with the `self.function.cls` check. Thanks!